### ISSUE TITLE
flake8-todos false negatives

### ISSUE THREAD

**Issue by JonathanPlasse** [charliermarsh/ruff#4549]
```python
# TODO
# ^^^^ Detect TODO errors
# foo # TODO
#       ^^^^ Do not detect TODO errors
```
On Ruff 0.0.269

**Comment by charliermarsh** [comment]
This _might_ be intentional. Do you have an example of this in the wild?

**Comment by evanrittenhouse** [comment]
Yeah, I picked the regexes to match the [existing plugin's regex](https://github.com/orsinium-labs/flake8-todos/blob/master/flake8_todos/_rules.py#L12) to avoid confusion on first iteration. If we'd like to change it, that's something that we can easily change.

**Comment by JonathanPlasse** [comment]
```python
class Foo:
    @property
    def id(  # noqa: A003 # TODO(jonathan): When doing BREAKING CHANGES rename to avoid shadowing builtin id
        self,
    ) -> str:
    pass

**Comment by evanrittenhouse** [comment]
Ah, you know what? It looks like they use `re.search`, meaning that the first location of the pattern is returned. @charliermarsh can you assign this to me please? I'll fix it
